### PR TITLE
Consolidate AWS session creation

### PR DESCRIPTION
**Pull request by sonofachamp**
Consolidate session creations.

#165 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/aws/session/session.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// FromProfile returns a session configured against the input profile name.
+func FromProfile(name string) (*session.Session, error) {
+	return session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+		Profile:           name,
+	})
+}
+
+// FromRole returns a session configured against the input role and region.
+func FromRole(roleARN string, region string) (*session.Session, error) {
```
Comment (efekarakus) [inline internal/pkg/aws/session/session.go]: Could we detect the region from the role's ARN? Using [arn.Parse](https://github.com/aws/aws-sdk-go/blob/v1.25.7/aws/arn/arn.go#L61) for example

Comment (sonofachamp) [reply internal/pkg/aws/session/session.go]: Hmm I don't think so, but I'm not certain. As far as my understanding goes there is no region in the role ARNs.

Comment (efekarakus) [reply internal/pkg/aws/session/session.go]: oh yeah, you're right!

Comment (kohidave) [reply internal/pkg/aws/session/session.go]: Love me some global resources :)

**Pull request merged**